### ISSUE TITLE
Infra flakes on "Linux flutter_packaging"

### ISSUE THREAD

**Issue by reidbaker** [flutter/flutter#126286]
https://ci.chromium.org/p/flutter/builders/prod/Linux%20flutter_packaging_test?limit=50

https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20flutter_packaging_test/796/overview

**Comment by ricardoamador** [comment]
The last few are timeouts. Can you look to see why packaging is taking so long?

**Comment by reidbaker** [comment]
in this infra flake I see 7.5 minutes of documented downloading 
and some download tasks that do not have timestamps or durations. What is the timeout for this test or where can I look it up myself. 
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20flutter_packaging_test/3868/overview

**Comment by reidbaker** [comment]
I will add that I see no error failures in the last 600 or so run only infra flakes for steps we know can be flakey like downloading.

**Comment by ricardoamador** [comment]
Hi @reidbaker I usually check here for the historical runs: https://ci.chromium.org/ui/p/flutter/builders. It shows all our builders in all pools.

It appears that if the test does not fail (timeout) it takes around 20 minutes. The prepare and create flutter archive, the step that is failing  takes ~17 minutes to complete.

In each case the step times out after the downloads when it is trying to prepare the tar ball. Not sure why that is taking so long.

**Comment by reidbaker** [comment]
Can you link to a specific instance?
https://ci.chromium.org/ui/p/flutter/builders shows me way more information
and I am popping into and out of this bug with other work.

**Comment by reidbaker** [comment]
@ricardoamador feel free to unassign yourself. This was just my way of sending this back to infra but I never assigned the bug to myself as i was investigating.

**Comment by ricardoamador** [comment]
@reidbaker circling back. No problem, we can leave it assigned as is.

**Comment by ricardoamador** [comment]
cc @reidbaker

Okay not sure if this is THE issue but I did notice that when the building of the archive is taking a long time causing the timeout it appears that there are other packages or directories that may have been left over from something else and not cleaned up properly. I took a look and compared the failing ones that time out they are adding a ton of other items when building the archive:

For example here is a build that passed and is a relatively short amount of time around the average of 20 or so minutes: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8761139945712791889/+/u/prepare_and_create_a_flutter_archive/stdout

And here is one from a build that timed out:
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8761169327645302481/+/u/prepare_and_create_a_flutter_archive/stdout

I did a quick comparison of the builds and what they are putting into the archive, and if you search for this string, "flutter/bin/cache/artifacts/," in both builds you will get about 220 items which looks to be the artifacts the test is expecting versus the failing build that shows 7,402 items all the way to the time limit, which includes fuchsia packages for different architectures which I am sure are not meant to be there.

So it looks like something is not cleaning up and leaving packages around that are getting picked up by this test and I need to figure out why this is.
